src/main/java/net/fabricmc/loom/configuration/providers/mappings/NoOpIntermediateMappingsProvider.java

```diff
@@ -29,19 +29,31 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 
+import org.gradle.api.Project;
+import org.gradle.api.provider.Property;
 import org.jetbrains.annotations.NotNull;
 
+import net.fabricmc.loom.LoomGradleExtension;
 import net.fabricmc.loom.api.mappings.intermediate.IntermediateMappingsProvider;
 
 /**
  * A bit of a hack, creates an empty intermediary mapping file to be used for mc versions without any intermediate mappings.
  */
 public abstract class NoOpIntermediateMappingsProvider extends IntermediateMappingsProvider {
 	private static final String HEADER = "tiny\t2\t0\tofficial\tintermediary";
+	private static final String HEADER_LEGACY = "tiny\t2\t0\tintermediary\tclientOfficial\tserverOfficial\t";
+
+	public Property<Boolean> legacy;
```

Comment:
As this class is already created via the object factory you can just do:

`public abstract Property<Boolean> getIsLegacy()`

This removes the need for you pass the Project through. You can likely remove the `configure` method, and set the convention for this in the extension where its created.

The `finalizeValueOnRead` can be set in a no args constructor marked with "@Inject"

Reply:
> As this class is already created via the object factory you can just do:
>
> `public abstract Property<Boolean> getIsLegacy()`
>
> This removes the need for you pass the Project through. You can likely remove the configure method, and set the convention for this in the extension where its created.

I had some trouble with that. I think that is what I tried first (because the other properties in the super class worked that way) but it wouldn't compile because of an abstract method error.

Reply:
`NoOpIntermediateMappingsProvider` has no child classes, so im not sure what issue you had. Maybe try again and let me know if you get stuck again?

Reply:
Alright. I'll push that and see if it works - it's very possible I was doing something wrong in my previous attempt.